### PR TITLE
Fixed build index deletion

### PR DESCRIPTION
**Pull request by stephenwf**
@giorgiosironi This will probably depend on the fix for the salt

### CODE REVIEW

**Comment by giorgiosironi** [comment]
Re-running once with new Salt formula before merging.

**Comment by giorgiosironi** [comment]
Some configuration for the option:
```
10:20:35 [54.146.170.47] out:           ID: search-ensure-index
10:20:35 [54.146.170.47] out:     Function: cmd.run
10:20:35 [54.146.170.47] out:         Name: ./bin/console search:setup --delete --env=ci
10:20:35 [54.146.170.47] out:       Result: False
10:20:35 [54.146.170.47] out:      Comment: Command "./bin/console search:setup --delete --env=ci" run
10:20:35 [54.146.170.47] out:      Started: 10:20:30.020274
10:20:35 [54.146.170.47] out:     Duration: 153.551 ms
10:20:35 [54.146.170.47] out:      Changes:   
10:20:35 [54.146.170.47] out:               ----------
10:20:35 [54.146.170.47] out:               pid:
10:20:35 [54.146.170.47] out:                   8956
10:20:35 [54.146.170.47] out:               retcode:
10:20:35 [54.146.170.47] out:                   1
10:20:35 [54.146.170.47] out:               stderr:
10:20:35 [54.146.170.47] out:                   
10:20:35 [54.146.170.47] out:                                                                             
10:20:35 [54.146.170.47] out:                     [Symfony\Component\Console\Exception\RuntimeException]  
10:20:35 [54.146.170.47] out:                     The "--delete" option requires a value.                 
10:20:35 [54.146.170.47] out:                                                                             
10:20:35 [54.146.170.47] out:                   
10:20:35 [54.146.170.47] out:                   search:setup [-d|--delete DELETE]
10:20:35 [54.146.170.47] out:               stdout:
```

File: src/Search/Api/Elasticsearch/Command/BuildIndexCommand.php [hunk 1]
```diff
@@ -77,5 +82,8 @@ protected function execute(InputInterface $input, OutputInterface $output)
         if ($create['payload'] instanceof SuccessResponse) {
             $logger->info('Created new index <comment>[Don\'t forget to re-index!]</comment>');
         }
+        if (isset($create['error'])) {
```
Comment (giorgiosironi) [inline src/Search/Api/Elasticsearch/Command/BuildIndexCommand.php]: I would go for `error` here

File: src/Search/Api/Elasticsearch/Command/BuildIndexCommand.php [hunk 2]
```diff
@@ -59,15 +63,16 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
         // Try removing old one.
-        try {
-            $delete = $this->client->deleteIndex();
-        } catch (Throwable $e) {
-            $logger->debug($e->getMessage(), $e->getTrace());
+        if ($toDelete) {
+            try {
+                $delete = $this->client->deleteIndex();
+            } catch (Throwable $e) {
+                $logger->debug($e->getMessage(), $e->getTrace());
```
Comment (giorgiosironi) [inline src/Search/Api/Elasticsearch/Command/BuildIndexCommand.php]: Standard way to log an exception during an important setup operation like this should be:
```
$logger->error("Cannot delete index", ['exception' => $e]);
```
Exceptions in the context can be dumped with the right level of verbosity (including stack traces) by log handlers.

**Comment by giorgiosironi** [comment]
Take a look at the comments for follow-up PRs

File: src/Search/Api/Elasticsearch/Command/BuildIndexCommand.php (hunk 2 above)
Comment (stephenwf) [inline src/Search/Api/Elasticsearch/Command/BuildIndexCommand.php]: The issue I've seen is on a clean environment where there is no index in the first instance, and an error is displayed saying "no index to delete"

If you're okay with the errors showing under those circumstances I can add it

**Pull request merged**